### DISCUSSION TITLE
General Instructions and manual

### DISCUSSION THREAD

**Discussion by robbmue** [hzi-braunschweig/SORMAS-data-generator#17] [Q&A]
Hi, 
since we've solved the issue I've had last week I didn't have much time to look further into the data generator. 
I do have some questions about the general usage and i hope to get some answers. 
If you could give me some bullet points i could work that into a small manual as well. 

- How do i control what kind of data is generated?
- How do i set the number of cases or contacts i want to generate?
- How do you import the data into a sormas instance other than the simple instance you've included? Or do i have to export/import the data?
- Which sormas versions are working for now?

Thanks in advance!

**Comment by robbmue** [comment]
Okay i guess to connect the generator to another instance i just start the container on the same machine with the same .env file, consider this answered.

**Reply by JonasCir** [reply]
Hi, if you checkout `development` and start docker-compose, a minimal SORMAS-Docker stack starts where the test data is automatically imported at startup. Once the data is imported, the importer container stops automatically.

**Comment by JonasCir** [comment]
Hi @robbmue ,
thank you for your questions and your offer:

> How do i control what kind of data is generated?

Not yet possible unfortunately. This will need some work (CC @stephaneghozzi )

> How do i set the number of cases or contacts i want to generate?

This will be implemented soon. I think this should be controlled via CLI or ENV vars.

> How do you import the data into a sormas instance other than the simple instance you've included? Or do i have to export/import the data?

The idea is to build the data-generator Docker image, push it to registry and build an docker-compose in SORMAS-Docker which includes the data generator (alternatively we could decide to include it in all docker-compose files and introduce an ENV switch which is read in `setup_and_run.sh` and is enabled on test/demo systems)

> Which sormas versions are working for now?

Works on all recent versions, however, we are dependent on API stability, so from time to time might need to fix some things before bumping to a newer version.

**Reply by JonasCir** [reply]
@robbmue If I understand correctly, you are associated with Netzlink. I got some questions regarding Docker image build/registry and demo instances. There is more and more time pressure to provide demo servers with sample data by default, so I would really appreciate if we could discuss further. :smiley:

**Comment by stephaneghozzi** [comment]
Sorry, just seeing the discussion now.

> How do i control what kind of data is generated?

@robbmue What do you mean? What would you like to control? E.g. the process how the data are generated, for which counties, the format in which they are stored, ...?

> How do i set the number of cases or contacts i want to generate?

I opened the issue https://github.com/hzi-braunschweig/SORMAS-data-generator/issues/19
